_posts/2019-12-10-variational-gaussian-process/vgp.Rmd

```diff
@@ -0,0 +1,549 @@
+
+> Gaussian process models in some sense bring together work in the two communities. 
+
+In this post, that "in some sense" gets very concrete. We'll see a Keras network, defined and trained the usual way, that has a Gaussian Process layer for its main constituent.
+The task will be "simple" multivariate regression.
+
+As an aside, this "bringing together communities" -- or ways of thinking, or solution strategies -- makes for a good overall characterization of TensorFlow Probability as well.
+^[So far, we have seen uses of TensorFlow Probability for such different applications as [adding uncertainty estimates to neural networks](https://blogs.rstudio.com/tensorflow/posts/2019-11-13-variational-convnet/), [Bayesian model estimation using Hamiltonian Monte Carlo](https://blogs.rstudio.com/tensorflow/posts/2019-05-24-varying-slopes/), or [linear-Gaussian state space models](https://blogs.rstudio.com/tensorflow/posts/2019-06-25-dynamic_linear_models_tfprobability/).]
+
+## Gaussian Processes 
+
+A Gaussian Process is a [distribution over functions, where the function values you sample are jointly Gaussian](https://math.stackexchange.com/a/2297480) - roughly speaking, a generalization to infinity of the multivariate Gaussian. Besides the reference book we already mentioned [@Rasmussen], there are a number of nice introductions on the net: see e.g. [https://distill.pub/2019/visual-exploration-gaussian-processes/](https://distill.pub/2019/visual-exploration-gaussian-processes/) or [https://peterroelants.github.io/posts/gaussian-process-tutorial/](https://peterroelants.github.io/posts/gaussian-process-tutorial/). And like on everything cool, there is a chapter on Gaussian Processes in the late David MacKay's [@MacKay] [book](http://www.inference.org.uk/itprnn/book.pdf).
+
+In this post, we'll use TensorFlow Probability's _Variational Gaussian Process_ (VGP) layer, designed to efficiently work with "big data". As Gaussian Process Regression (GPR, from now on) involves the inversion of a -- possibly big -- covariance matrix, attempts have been made to design approximate versions, often based on variational principles. The TFP implementation is based on papers by Titsias (2009) [@titsias09a] and Hensman et al. (2013) [@HensmanFL13]. Instead of $p(y|X)$, the actual probability of the target data given the actual input, we work with a variational distribution $q(u)$ that acts as a lower bound. 
+
+Here $u$ is a set of so-called  _inducing index points_  specified by the user, chosen to well cover the range of the actual data. This algorithm is a lot faster than "normal" GPR, as only the covariance matrix of $u$ has to be inverted. As we'll see below, at least in this example (as well as in others not described here) it seems to be pretty robust as to the number of _inducing points_ passed.
```

Comment:
Is `u` the index points or is it the function (as in the `f` in `y = f(X) + noise`) values at the index points?

Reply:
right, great you spotted that! Thanks!

Reply:
looking again at that paragraph, i should probably have some `\mathbf`'s there as well...